ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -327,6 +328,18 @@ extern "C" void *__ykllvmwrap_irtrace_compile(char *FuncNames[], size_t BBs[],
             Value *Alloca = Builder.CreateAlloca(
                 OpTy->getPointerElementType(), OpTy->getPointerAddressSpace());
             VMap[Op] = Alloca;
+          } else if (isa<GlobalVariable>(Op)) {
+            // If there's a reference to a GlobalVariable, copy it over to the
+            // new module.
+            GlobalVariable *OldGV = cast<GlobalVariable>(Op);
+            GlobalVariable *GV = new GlobalVariable(
```

Comment:
Should we check if the global is already defined and skip this step if so? Perhaps we need a test where we trace two separate functions that reference the same global.

Reply:
Very good point. We definitely need a test for that. I'll try to construct one.